### ISSUE TITLE
Question: Move db based configuration to yml?

### ISSUE THREAD

**Issue by wernerkrauss** [Firesphere/silverstripe-newsmodule#138]
Is there a need for using Database just for configuration? Why not use _config.yml for most static stuff?

From my point of view an admin has access to yml and a customer shouldn't play with settings.

What's your opinion on this?

Pros: more speed, less DB calls, as config is cached automatically.

**Comment by wernerkrauss** [comment]
Well, i love config in yml, cause you can put it to git and deploy with your project. I hate when i have to click around to get settings for a new website running. But that may be a matter of taste.

Looking again at it: IMHO everything in $db (https://github.com/Firesphere/silverstripe-newsmodule/blob/master/code/extensions/NewsSiteConfigExtension.php#L16) can be in yml, the default images should be in DB (as it's a relation to a file and might change) but could also be set in the NewsPage (again matter of taste and not that important to me).

The only big benefit of having it in db: you can set different settings for each language. But this would be easy in yml, too.

```yml
News:
  my-value:
    - 'foo' #default value (or maybe prefix with default)
    - en_US: 'english value'
    - de_AT: 'austrian value'  
```

actions can be translated like in my concept: https://github.com/wernerkrauss/silverstripe-translatable-actions

(though translatable config values should go to lang files, too... _t() is just the right tool for that)
Of course we'd have to provide at least good docs for upgrading, maybe a simple tool?

**Comment by Firesphere** [comment]
I agree, most should be transferred to YML. PR's welcome. I'm lacking time to actually work on this right now :(
